src/main/java/org/variantsync/diffdetective/diff/GitDiffer.java

```diff
@@ -308,18 +307,26 @@ private static CommitDiffResult getPatchDiffs(
 
                 diffFormatter.format(diffEntry);
                 final String gitDiff = outputStream.toString(StandardCharsets.UTF_8);
-                final Result<PatchDiff, DiffError> patchDiff =
-                        getBeforeFullFile(git, parentCommit, diffEntry.getOldPath()).unwrap()
-                        .bind(file -> createPatchDiff(
-                                commitDiff,
-                                diffEntry,
-                                gitDiff,
-                                file,
-                                parseOptions).unwrap()
+                final String filename = diffEntry.getOldPath();
+                try {
+                    final BufferedReader file = getBeforeFullFile(git, parentCommit, filename);
+                    final PatchDiff patchDiff =
+                        createPatchDiff(
+                            commitDiff,
+                            diffEntry,
+                            gitDiff,
+                            file,
+                            parseOptions
                         );
 
-                patchDiff.ifSuccess(commitDiff::addPatchDiff);
-                patchDiff.ifFailure(errors::add);
+                    commitDiff.addPatchDiff(patchDiff);
+                } catch (IOException e) {
+                    Logger.debug(e, "Could not obtain full diff of file " + filename + " before commit " + parentCommit + "!");
+                    errors.add(DiffError.COULD_NOT_OBTAIN_FULLDIFF);
```

Comment:
Compared to before, I do think this is better. Most error constructors didn't really store the error message. They just logged it and forgot about it. When I first read these methods I was really confused why I have to pass a message into a function which just logs it for me.
The main thing I like to change in the future is to provide the filename and the commit hash inside of the error, so that the caller can decide what to do with them. This should removed the need for log & throw.

Reply:
Note there still are such "just log the message" smart constructors. See `CommitDiffResult` below.

Reply:
Ok